src/Sema.zig

```diff
@@ -19553,6 +19553,9 @@ fn analyzeSlice(
             if (!end_is_len) {
                 const end = try sema.coerce(block, Type.usize, uncasted_end_opt, end_src);
                 if (try sema.resolveMaybeUndefVal(block, end_src, end)) |end_val| {
+                    if (end_val.compare(.gt, len_val, Type.usize)) {
+                        return sema.fail(block, end_src, "out of bounds slice", .{});
```

Comment:
Can we match the corresponding array message? "index {d} out of bounds of slice of length {d} (+1 sentinel)"

Reply:
I'm sure we can, might wait for Vexu's #11253 which I'm guessing you'll look at today. That way we can use that since we have the type.

Reply:
Merged!